runtime/test/akms-test.js

```diff
@@ -0,0 +1,84 @@
+describe('arcs key management', function() {
+    describe('KeyManager', () => {
+        it('supports generating wrapped Storage keys from DeviceKeys, and rewrapping them', async () => {
+            const keyGenerator = KeyManager.getGenerator();
+            const deviceKey = await keyGenerator.generateDeviceKey();
+            const cloudKey = await keyGenerator.generateDeviceKey();
+
+            const storageKey = await KeyManager.getGenerator().generateWrappedStorageKey(deviceKey);
+            assert.isNotNull(storageKey);
+
+            const rewrappedKey = await storageKey.rewrap(deviceKey.privateKey(), cloudKey.publicKey());
+            assert.isNotNull(rewrappedKey);
+
+            // Test session key
+            const sessionKey = await rewrappedKey.unwrap(cloudKey.privateKey());
+            const testBuffer = new Uint8Array([42, 23, 99]);
+            const encryptBuffer = await sessionKey.encrypt(testBuffer);
+            const decryptBuffer = await sessionKey.decrypt(encryptBuffer);
+            assert.deepEqual(testBuffer, new Uint8Array(decryptBuffer));
+        });
+    });
+
+    describe('KeyStorage', () => {
+        it('supports persisting DeviceKeys and wrapped Storage Keys', async () => {
+            const keyGenerator = KeyManager.getGenerator();
+            const deviceKey = await keyGenerator.generateDeviceKey();
+            const keyStorage = KeyManager.getStorage();
+
+            const storageKey = await KeyManager.getGenerator().generateWrappedStorageKey(deviceKey);
+            assert.isNotNull(storageKey);
+
+            await keyStorage.write(deviceKey);
+            const foundKey = await keyStorage.find(await deviceKey.fingerprint());
+            assert(await deviceKey.fingerprint(), await foundKey.fingerprint());
+
+            await keyStorage.write(storageKey);
+            const foundStorageKey = await keyStorage.find(await storageKey.fingerprint());
+            assert(await foundStorageKey.fingerprint(), await storageKey.fingerprint());
+
+            const sessionKey = await storageKey.unwrap(deviceKey.privateKey());
+            const foundSessionKey = await foundStorageKey.unwrap(foundKey.privateKey());
+            const testBuffer = new Uint8Array([42, 23, 99]);
+            const encryptBuffer = await sessionKey.encrypt(testBuffer);
+            const decryptBuffer = await foundSessionKey.decrypt(encryptBuffer);
+            assert.deepEqual(testBuffer, new Uint8Array(decryptBuffer));
+        });
+    });
+
+    it('supports importing GCP PEM keys and exporting a rewrapped key', async () => {
+        const keyGenerator = KeyManager.getGenerator();
+        const deviceKey = await keyGenerator.generateDeviceKey();
+
+        const storageKey = await KeyManager.getGenerator().generateWrappedStorageKey(deviceKey);
+        assert.isNotNull(storageKey);
+
+        const pemKey = await Fetch('https://cloud-certs.storage.googleapis.com/google-cloud-csek-ingress.pem').then(resp => resp.text());
```

Comment:
Unit tests probably shouldn't fetch, so maybe we need a copy locally or somehow mock/fake?

Reply:
Or could move this test case to a manual-test for now.

Reply:
moved to manual for now.

Reply:
Moved back to automatic but removed the fetch dependency, just hard coded the test cert